infrastructure/kube/templates/relay-maintainer/initcontainer/provision-relay-maintainer.js

```diff
@@ -0,0 +1,115 @@
+is long wait times for scripts to execute.
+*/
+const web3_options = {
+  defaultBlock: 'latest',
+  defaultGas: 4712388,
+  transactionBlockTimeout: 25,
+  transactionConfirmationBlocks: 3,
+  transactionPollingTimeout: 480
+}
+
+const web3 = new Web3(contractOwnerProvider, null, web3_options)
+
+async function provisionRelayMaintainer() {
+  console.log('###########  Provisioning relay maintainer! ###########')
+
+  console.log(`\n<<<<<<<<<<<< Read operator address from key file >>>>>>>>>>>>`)
+  const operatorAddress = readAddressFromKeyFile(operatorKeyFile)
+
+  console.log(`\n<<<<<<<<<<<< Funding Operator Account ${operatorAddress} >>>>>>>>>>>>`)
+  await fundOperator(operatorAddress, purse, '10')
+
+  console.log('\n<<<<<<<<<<<< Creating relay maintainer env config file >>>>>>>>>>>>')
+  await createRelayMaintainerConfig()
+
+  console.log("\n########### keep-ecdsa Provisioning Complete! ###########")
+}
+
+function readAddressFromKeyFile(keyFilePath) {
+  const keyFile = JSON.parse(fs.readFileSync(keyFilePath, 'utf8'))
+
+  return web3.utils.toHex(keyFile.address)
+}
+
+async function fundOperator(operatorAddress, purse, requiredEtherBalance) {
+  let requiredBalance = web3.utils.toBN(web3.utils.toWei(requiredEtherBalance, 'ether'))
+
+  const currentBalance = web3.utils.toBN(await web3.eth.getBalance(operatorAddress))
+  if (currentBalance.gte(requiredBalance)) {
+    console.log(`Operator address is already funded, current balance: ${web3.utils.fromWei(currentBalance)}`)
+    return
+  }
+
+  const transferAmount = requiredBalance.sub(currentBalance)
+
+  console.log(`Funding account ${operatorAddress} with ${web3.utils.fromWei(transferAmount)} ether from purse ${purse}`)
+  await web3.eth.sendTransaction({ from: purse, to: operatorAddress, value: transferAmount })
+  console.log(`Account ${operatorAddress} funded!`)
+}
+
+async function createRelayMaintainerConfig() {
+  const envTemplate = fs.readFileSync('/tmp/env-template', 'utf8')
+
+  const operatorKey = readKeyFromKeyFile(keyFilePath)
+  const relayContractAddress = RelayJSON.networks[ethNetworkId].address
+  const ethURL = new URL(process.env.ETH_RPC_URL)
+
+  const ethNetworkName = ""
+  const bcoinHost = ""
+  const bcoinPort = ""
+  const bcoinApiKey = ""
+  const infuraKey = ""
+
+  const finalEnv =
+    envTemplate
+      .replace(/^(SUMMA_RELAY_ETHER_HOST=).*$/, `\\1${ethURL.hostname}`)
+      .replace(/^(SUMMA_RELAY_ETHER_PORT=).*$/, `\\1${ethURL.port}`)
+      .replace(/^(SUMMA_RELAY_OPERATOR_KEY=).*$/, `\\1${operatorKey}`)
+      .replace(/^(SUMMA_RELAY_ETH_NETWORK=).*$/, `\\1${ethNetworkName}`)
+      .replace(/^(SUMMA_RELAY_ETH_CHAIN_ID=).*$/, `\\1${ethNetworkId}`)
+      .replace(/^(SUMMA_RELAY_BCOIN_HOST=).*$/, `\\1${bcoinHost}`)
+      .replace(/^(SUMMA_RELAY_BCOIN_PORT=).*$/, `\\1${bcoinPort}`)
+      .replace(/^(SUMMA_RELAY_BCOIN_API_KEY=).*$/, `\\1${bcoinApiKey}`)
+      .replace(/^(SUMMA_RELAY_INFURA_KEY=).*$/, `\\1${infuraKey}`)
```

Comment:
I'm not sure what this is.

Reply:
We need to get a real Infura set up heh. Infura lets you require an API key alongside your requests so that others can't use your Infura info, basically.

Reply:
Do we need to do this before we try deploying?